packages/webapp/src/components/Animals/Inventory/index.tsx

```diff
@@ -14,44 +14,95 @@
  */
 import Table from '../../../components/Table';
 import Layout from '../../../components/Layout';
+import PureSearchBarWithBackdrop from '../../PopupFilter/PureSearchWithBackdrop';
+import NoSearchResults from '../../../components/Card/NoSearchResults';
 import type { AnimalInventory } from '../../../containers/Animals/Inventory/useAnimalInventory';
 import { TableV2Column, TableKind } from '../../Table/types';
-import type { DefaultTheme } from '@mui/styles';
+import type { Dispatch, SetStateAction } from 'react';
+import styles from './styles.module.scss';
+import clsx from 'clsx';
+
+export type SearchProps = {
+  searchString: string | null | undefined;
+  setSearchString: Dispatch<SetStateAction<string[]>>;
+  placeHolderText: string;
+  searchResultsText: string;
+};
 
 const PureAnimalInventory = ({
-  tableData,
+  filteredInventory,
   animalsColumns,
-  theme,
-  isMobile,
+  zIndexBase,
+  backgroundColor,
+  isDesktop,
+  searchProps,
 }: {
-  tableData: AnimalInventory[];
+  filteredInventory: AnimalInventory[];
   animalsColumns: TableV2Column[];
-  theme: DefaultTheme;
-  isMobile: boolean;
+  zIndexBase: number;
+  backgroundColor: string;
+  isDesktop: boolean;
+  searchProps: SearchProps;
 }) => {
+  const { searchString, setSearchString, placeHolderText, searchResultsText } = searchProps;
+  const hasSearchResults = filteredInventory.length !== 0;
+
   return (
     <Layout
       classes={{
         container: {
-          backgroundColor: theme.palette.background.paper,
-          borderRadius: '8px',
-          border: '1px solid var(--Colors-Primary-Primary-teal-50)',
-          marginTop: '16px',
+          backgroundColor: backgroundColor,
+          borderRadius: isDesktop && '8px',
+          border: isDesktop && '1px solid var(--Colors-Primary-Primary-teal-50)',
+          marginTop: isDesktop && '16px',
+          padding: !isDesktop ? '0px' : '16px',
         },
       }}
       hasWhiteBackground
       footer={false}
     >
-      <Table
-        kind={TableKind.V2}
-        alternatingRowColor={true}
-        columns={animalsColumns}
-        data={tableData}
-        shouldFixTableLayout={!isMobile}
-        minRows={tableData.length}
-        dense={false}
-        showHeader={!isMobile}
-      />
+      <div
+        className={clsx(
+          isDesktop ? styles.SearchAndFilter_container___desktop : styles.SearchAndFilter_container,
+        )}
+      >
+        <PureSearchBarWithBackdrop
+          value={searchString}
+          onChange={(e: any) => setSearchString(e.target.value)}
+          isSearchActive={!!searchString}
+          placeholderText={placeHolderText}
+          zIndexBase={zIndexBase}
+          isDesktop={isDesktop}
+          className={clsx(isDesktop ? styles.SearchBar___widthFixed : styles.SearchBar___widthFull)}
+        />
+        <div
+          className={clsx(
+            isDesktop ? styles.SearchResults___marginLeft8 : styles.SearchResults___textAlignCenter,
```

Comment:
I'd try to give these classes more generic names rather than names that specifically point out the styling applied. The reason for that is, it leaves little flexibility to update the styling and reuse that selector without having to change the name of the class. Even if we went for BEM, the "modifiers" used are usually things like "compact", "disabled", "active", "big" rather than calling out specific styles

Reply:
Yeah. I would only call out specific styles if it is the only style on the property. I don't see value in making the modifier less readable (in JS) when there is just one style. 

But yeah anyone wanting to add another style would probably rename it to be more generic. Most just showcasing the inline breakpoints readability -- I haven't mastered BEM by any means ahha but it is nice